math/mathcore/src/TStatistic.cxx

```diff
@@ -56,80 +60,90 @@ TStatistic::~TStatistic()
 /// \param[in] val Value to fill the Tstatistic with
 /// \param[in] w The weight of the value
 ///
-/// Also updates statistics in the object. For number of entries, sum of weights,
-/// sum of squared weights and sum of (value*weight), one extra value is added to the
-/// statistic. For the sum of squared (value*weight) pairs, the function uses formula 1.4
-/// in Chan-Golub, LeVeque : Algorithms for computing the Sample Variance (1983),
-/// genralized by LM for the case of weights:
+/// Also updates statistics in the object. For number of entries, sum of entry
+/// values, sum of weights, sum of squared weights and sum of (value*weight),
+/// one extra value is added to the statistic. The sum of squared
+/// (value*weight) pairs is updated using formula 1.4 in Chan-Golub, LeVeque :
+/// Algorithms for computing the Sample Variance (1983), generalized by LM for
+/// the case of weights
 /// \f[
 ///   \frac{w_j}{\sum_{i=0}^{j} w_i \cdot \sum_{i=0}^{j-1} w_i} \cdot
-///   \left(
-///         \sum_{i=0}^{j} w_i \cdot val_i -
+///   \left[
+///         \left(\sum_{i=0}^{j} w_i\right) \cdot val_j -
 ///         \sum_{i=0}^{j} \left(w_i \cdot val_i\right)
-///   \right)
+///   \right]
 /// \f]
 ///
-/// The minimum(maximum) is computed by checking that the fill value
-/// is either less(greater) than the current minimum(maximum)
+/// The minimum(maximum) is computed by checking that the fill value is either
+/// less(greater) than the current minimum(maximum).
 void TStatistic::Fill(Double_t val, Double_t w) {
 
 
    if (w == 0) return;
-   // increase data count
+   // Increase data count
    fN++;
 
-   // update sum of weights
+   // Current sum of weights
    Double_t tW = fW + w;
 
-   // update sum of (value * weight) pairs
+   // Update sum of (value * weight) pairs
    fM += w * val;
 
-   // update minimum and maximum values
+   // Update minimum and maximum values
    fMin = (val < fMin) ? val : fMin;
    fMax = (val > fMax) ? val : fMax;
 
-//      Double_t dt = val - fM ;
+   // Check sum of weights
    if (tW == 0) {
       Warning("Fill","Sum of weights is zero - ignore current data point");
       fN--;
       return;
    }
 
-   if (fW != 0) {  // from the second time
+   if (fW != 0) {  // From the second time
       Double_t rr = ( tW * val - fM);
       fM2 += w * rr * rr / (tW * fW);
    }
+   // Update sum of weights in the TStatistic object
    fW = tW;
+   // Update sum of weights squared
    fW2 += w*w;
+   // Update sum of values
+   fSum += val;
```

Comment:
do you really want to ignore the weight?

Reply:
There is already the weighted sum in the class data members, it's just not printed. Maybe I could refactor the PR in order to print the weighted sum when the user also includes weights as arguments to the constructor, or just the simple sum if there are only the values